src/index.ts

```diff
@@ -72,23 +72,45 @@ export default function wrapMotoko(compiler: Compiler) {
     };
 
     // Function signatures for `mo.parseMotokoTyped()`
-    type ParseMotokoTypedResult = { ast: Node; type: Node };
-    function parseMotokoTyped(paths: string): ParseMotokoTypedResult;
-    function parseMotokoTyped(paths: string[]): ParseMotokoTypedResult[];
+    type ParseMotokoTypedResult = {
+        ast: Node;
+        type: Node;
+        immediateImports: string[];
+    };
+    function parseMotokoTyped(
+        paths: string,
+        scopeCache: Map<string, Scope>,
+    ): [ParseMotokoTypedResult, Map<string, Scope>];
+    function parseMotokoTyped(
+        paths: string[],
+        scopeCache: Map<string, Scope>,
+    ): [ParseMotokoTypedResult[], Map<string, Scope>];
```

Comment:
Can we make this backwards-compatible with a default `scopeCache` argument?

```suggestion
    function parseMotokoTyped(
        paths: string,
        scopeCache?: Map<string, Scope>,
    ): [ParseMotokoTypedResult, Map<string, Scope>];
    function parseMotokoTyped(
        paths: string[],
        scopeCache?: Map<string, Scope>,
    ): [ParseMotokoTypedResult[], Map<string, Scope>];
```